### PR TITLE
Adds the anti-teleportation implant to the game/the space hotel

### PR DESCRIPTION
**Pull request by ATH1909**
## About The Pull Request

The anti-teleportation implant is a currently uncraftable/unprintable subtype of the exile implant that, as the name implies, keeps you from teleporting.

Space hotel staff (and space hotel sec) will now spawn with anti-teleportation implants.

## Why It's Good For The Game

This implant will help the staff of the space hotel resist the urge to use the space hotel's incredibly alluring roundstart teleporter to ignore their flavor/greeting text and come to the station.

And yes, I know that we currently have the space hotel space ruin disabled, but maybe this change could lead to the re-enabling of that space ruin in the future.

## Changelog
:cl: ATHATH and Rohesie
add: Slavers (and particularly strict employers) have begun using anti-teleportation implants to keep their slaves (and wagies, respectively) from using teleporters to escape their grasp.
add: The staff (and sec force) of the currently-disabled (on /tg/) space hotel ruin now spawn with anti-teleportation implants.
/:cl:

### CODE REVIEW

**Comment by Pugie** [comment]
YOU CAN CHECK OUT ANYTIME YOU LIKE
BUT YOU CAN'T EVER LEAAAAAAAAAAAVE

**Comment by GravitronBlue** [comment]
Might want to think about preventing hotel staff from being able to remove eachother's implants assuming ghetto surgery is viable.

**Comment by ATH1909** [comment]
> Might want to think about preventing hotel staff from being able to remove eachother's implants assuming ghetto surgery is viable.

Eh, you have to go significantly out of your way to do that, which means that you can just be bwoinked. This is just to keep impulse decisions in check. It's sort of like how comms agents technically could use trickery to cross their lava moat without dying, but doing so isn't practical because the admins will just smite you.

**Comment by GravitronBlue** [comment]
> This is just to keep impulse decisions in check.

Fair enough for now unless things change, as usually it's just lone individuals that abuse the hotel staff role to go back to the station.

Might still take a look around and make sure there aren't other cheap ways to escape like the EVA spacesuit (assuming it still exists), or other issues like #14540.

**Comment by ATH1909** [comment]
> > This is just to keep impulse decisions in check.
> 
> Fair enough for now unless things change, as usually it's just lone individuals that abuse the hotel staff role to go back to the station.
> 
> Might still take a look around and make sure there aren't other cheap ways to escape like the EVA spacesuit (assuming it still exists), or other issues like #14540.

I'd prefer to not make it IMPOSSIBLE to escape, in case the hotel employees somehow get culted or revved or given a greentext book or something (despite their mindshield implants).

File: code/game/objects/items/implants/implant_exile.dm [hunk 1]
```diff
@@ -4,19 +4,54 @@
...
+	desc = "Uses impressive bluespace grounding techniques to deny the person implanted by this implant the ability to teleport (or be teleported). Used by certain slavers (or particularly strict employers) to keep their slaves from using teleporters to escape their grasp."
+
+/obj/item/implant/exile/noteleport/get_data()
+	var/dat = {"<b>Implant Specifications:</b><BR>
+				<b>Name:</b> Anti-Teleportation Implant<BR>
+				<b>Implant Details:</b> Keeps the implantee from using most teleportation devices. In addition, it spoofs the implant signature of an exile implant to keep the implantee from using certain gateway systems.<BR>"}
+	return dat
+
+/obj/item/implant/exile/noteleport/implant(mob/living/target, mob/user, silent = FALSE, force = FALSE)
+	if(..() && isliving(target))
```
Comment (Rohesie) [inline code/game/objects/items/implants/implant_exile.dm:28]:
```suggestion
/obj/item/implant/exile/noteleport/implant(mob/living/target, mob/user, silent = FALSE, force = FALSE)
	. = ..()
	if(!. || !isliving(target))
		return FALSE
```

File: code/game/objects/items/implants/implant_exile.dm [hunk 2]
```diff
@@ -4,19 +4,54 @@
...
+		var/mob/living/L = target
+		ADD_TRAIT(L, TRAIT_NO_TELEPORT, "implant")
+		return TRUE
+	return FALSE
```
Comment (Rohesie) [inline code/game/objects/items/implants/implant_exile.dm:32]:
```suggestion
	var/mob/living/living_target = target
	ADD_TRAIT(living_target, TRAIT_NO_TELEPORT, "implant")
	return TRUE
```
Better var names.

File: code/game/objects/items/implants/implant_exile.dm [hunk 3]
```diff
@@ -4,19 +4,54 @@
...
+
+/obj/item/implant/exile/noteleport/removed(mob/target, silent = FALSE, special = FALSE)
+	if(..() && isliving(target))
+		var/mob/living/L = target
+		REMOVE_TRAIT(L, TRAIT_NO_TELEPORT, "implant")
+		return TRUE
+	return FALSE
```
Comment (Rohesie) [inline code/game/objects/items/implants/implant_exile.dm:39]:
```suggestion
/obj/item/implant/exile/noteleport/removed(mob/target, silent = FALSE, special = FALSE)
	. = ..()
	if(!. || !isliving(target))
		return FALSE
	var/mob/living/living_target = target
	REMOVE_TRAIT(living_target, TRAIT_NO_TELEPORT, "implant")
	return TRUE
```

**Comment by Rohesie** [comment]
Do you still plan to work on this?

**Comment by ATH1909** [comment]
Yes, hold on.

**Comment by ATH1909** [comment]
Done. You've been credited in the changelog as well.

**Comment by LemonInTheDark** [comment]
`TRAIT_NO_TELEPORT` hurts my soul (CHECK CONTENTS YOU BASTARD), and it really would be best if they had no good way to get the good old taser shotgun on station, but this isn't that, and it's a good first step. We shouldn't ask the headmins to re-enable after just this, cause again, taser shotgun, but nothing about this seems seriously wrong. Good for our one downstream who uses the map too.

**Pull request merged**